### PR TITLE
Add inlay hints documentation

### PR DESCRIPTION
**Pull request by SomeoneToIgnore**
Closes https://github.com/rust-analyzer/rust-analyzer/issues/3267

### CODE REVIEW

File: docs/user/features.md [hunk 1]
```diff
@@ -156,3 +156,28 @@ default highlighter.
 
 Experimental feature that, given code highlighting using rust-analyzer is
 active, will pick unique colors for identifiers.
+
+### Code hints
+
+Rust Analyzer has two types of hints to show the information about the code:
```
Comment (flodiebold) [inline docs/user/features.md]:
```suggestion
Rust-analyzer has two types of hints to show the information about the code:
```

File: docs/user/features.md [hunk 2]
```diff
@@ -156,3 +156,28 @@ default highlighter.
...
+
+* hover hints, appearing on hover on any element.
+
+Contains extended information on the language item hovered.
```
Comment (flodiebold) [inline docs/user/features.md]:
```suggestion
These contain extended information on the hovered language item.
```

File: docs/user/features.md [hunk 3]
```diff
@@ -156,3 +156,28 @@ default highlighter.
...
+
+* type hints, displaying the minimal information on the type of the expression (if the information is available)
+* parameter name hints, displaying the names of the parameters in the corresponding methods
+
+#### VS Code
+
+In VS Code, the following settings can be used to configure the inlay hints:
+
+* rust-analyzer.displayInlayHints — toggles inlay hints display on or off
+* rust-analyzer.maxInlayHintLength — shortens the hints if their length exceed the value specified. If no value is specified (`null`), no shorteninng is applied.
```
Comment (flodiebold) [inline docs/user/features.md]:
```suggestion
* rust-analyzer.maxInlayHintLength — shortens the hints if their length exceeds the value specified. If no value is specified (`null`), no shorteninng is applied.
```

File: docs/user/features.md [hunk 4]
```diff
@@ -156,3 +156,28 @@ default highlighter.
...
+
+**Note:** VS Code does not have a native support for inlay hints [yet](https://github.com/microsoft/vscode/issues/16221) and the hints are implemented using decorations.
```
Comment (flodiebold) [inline docs/user/features.md]:
```suggestion
**Note:** VS Code does not have native support for inlay hints [yet](https://github.com/microsoft/vscode/issues/16221) and the hints are implemented using decorations.
```

File: docs/user/features.md [hunk 5]
```diff
@@ -156,3 +156,28 @@ default highlighter.
...
+Two types of inlay hints are displayed currently:
+
+* type hints, displaying the minimal information on the type of the expression (if the information is available)
+* parameter name hints, displaying the names of the parameters in the corresponding methods
+
+#### VS Code
+
+In VS Code, the following settings can be used to configure the inlay hints:
+
+* rust-analyzer.displayInlayHints — toggles inlay hints display on or off
```
Comment (lnicola) [inline docs/user/features.md]:
```suggestion
* `rust-analyzer.displayInlayHints` — toggles inlay hints display on or off
```

File: docs/user/features.md (hunk 3 above)
Comment (lnicola) [inline docs/user/features.md]:
```suggestion
* `rust-analyzer.maxInlayHintLength` — shortens the hints if their length exceed the value specified. If no value is specified (`null`), no shortening is applied.
```

**Comment by lnicola** [comment]
shorteninng? :disappointed:

**Comment by flodiebold** [comment]
bors cancel

**Comment by SomeoneToIgnore** [comment]
Thanks for spotting it all.

**Pull request merged**